File: testdata/bench/cpp/BenchmarkTest00089.cpp
#include <cstdio>
#include <cstdlib>

int main() {
    char *url = getenv("QUERY_STRING");
    printf("Content-Type: text/html\r\n\r\n");
    printf("<html><body><a href='%s'>Click</a></body></html>", url);
    return 0;
}

File: testdata/bench/rust/BenchmarkTest00089.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: XSS via variable indirection
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("content").unwrap_or(&String::new()).clone();
    let x = param.clone();
    let html = format!("<p>{}</p>", x);
    HttpResponse::Ok().content_type("text/html").body(html)
}
